docs/examples/userguide/extension_types/dataclass.py

```diff
@@ -0,0 +1,21 @@
+import cython
+try:
+    import typing
+    import dataclasses
+except ImportError:
+    pass  # The modules don't actually have to exists for Cython to use them as annotations
+
+@cython.dataclasses.dataclass
+@cython.cclass
+class MyDataclass:
+    # fields can be declared using annotations
+    a: cython.int = 0
+    b: double = cython.dataclasses.field(default_factory = lambda: 10, repr=False)
+
+
+    c: str = 'hello'
```

Comment:
I'm not sure what the best way to present this is. The only line in the old-style syntax was:

```
# fields can also be declared using `cdef
cdef str c
c = "hello"  # assignment of default value on a separate line
```

with every other line in the new-style syntax. The way it was written was designed to show both in a single example, which I think is probably fine?

Reply:
> In the example also constructs cimport and cdef class are used

You're right - I missed those.

I think with Cython dataclasses we're expecting people to use annotation syntax (even in "Cython" mode) just because that's closer to Python dataclasses. So I really only wanted a quick note to say "you can use cdef attributes if you like"

It's probably fine as you have it though - I don't feel too strongly about it.